modules/apps/oauth-client/oauth-client-persistence-test/src/testIntegration/java/com/liferay/oauth/client/persistence/service/test/OAuthClientEntryLocalServiceTest.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.oauth.client.persistence.service.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.oauth.client.persistence.constants.OAuthClientEntryConstants;
+import com.liferay.oauth.client.persistence.exception.DuplicateOAuthClientEntryException;
+import com.liferay.oauth.client.persistence.exception.OAuthClientEntryAuthRequestParametersJSONException;
+import com.liferay.oauth.client.persistence.exception.OAuthClientEntryAuthServerWellKnownURIException;
+import com.liferay.oauth.client.persistence.exception.OAuthClientEntryOIDCUserInfoMapperJSONException;
+import com.liferay.oauth.client.persistence.model.OAuthClientEntry;
+import com.liferay.oauth.client.persistence.service.OAuthClientEntryLocalService;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.test.rule.DataGuard;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Christian Moura
+ */
+@DataGuard(scope = DataGuard.Scope.METHOD)
+@RunWith(Arquillian.class)
+public class OAuthClientEntryLocalServiceTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@BeforeClass
+	public static void setUpClass() {
+		_infoJSON = JSONUtil.put(
+			"client_id", RandomTestUtil.randomString()
+		).put(
+			"client_name", "Client to Google"
+		).put(
+			"client_secret", RandomTestUtil.randomString()
+		).put(
+			"grant_types",
+			JSONUtil.putAll("authorization_code", "refresh_token")
+		).put(
+			"response_types", JSONUtil.putAll("code")
+		).put(
+			"scope", "openid email profile"
+		).toString();
+	}
+
+	@Test
+	public void testAddDuplicatedOAuthClientEntry() throws Exception {
```

Comment:
Hey @cdbm, apply this rule here https://github.com/brianchandotcom/liferay-portal/commit/4e05e27682df48ead6349bbde92e15a96da41238